sunbeam-python/sunbeam/core/steps.py

```diff
@@ -125,22 +109,29 @@ def get_accepted_application_status(self) -> list[str]:
     )
     def run(self, status: Status | None = None) -> Result:
         """Apply terraform configuration to deploy sunbeam machine."""
-        machine_ids: list[str] = []
-
-        try:
-            app = self.jhelper.get_application(self.application, self.model)
-            machine_ids.extend(unit.machine for unit in app.units.values())
-        except ApplicationNotFoundException as e:
-            LOG.debug(str(e))
-
         try:
             extra_tfvars = self.extra_tfvars()
-            extra_tfvars.update(
-                {
-                    "machine_ids": machine_ids,
-                    "machine_model": self.model,
-                }
-            )
+            extra_tfvars["machine_model"] = self.model
+            if "machine_ids" not in extra_tfvars:
+                machine_ids: set[str] = set()
+                nodes: list[dict] = []
+
+                for role in self.roles:
+                    if isinstance(role, Role):
+                        role = [role]
+                    nodes = self.client.cluster.list_nodes_by_role(
+                        roles_to_str_list(role)
+                    )
+                    machine_ids.update(
+                        {
+                            node["machineid"]
+                            for node in nodes
+                            if node.get("machineid", -1) != -1
+                        }
+                    )
+
+                extra_tfvars["machine_ids"] = sorted(machine_ids)
+
```

Comment:
Don't we need to better manage timeouts in this case?

I wonder if in MAAS mode, deploying that many app won't cause an issue?

Reply:
Since this is part of the terraform apply, we need to be wary of the terraform apply timeout.

After 1200 seconds, it just kills terraform, which leaves the state in shamble...

Reply:
Terraform plan will return sooner, increased machine application timeouts to 30 minutes